doc/vlan_stacking/vlan_stacking_HLD.md

```diff
@@ -0,0 +1,635 @@
+
+```json
+"VLAN_STACKING": {
+    "Ethernet30|egress|20": {
+        "action": "swap",
+        "s_vlanid": "10"
+}
+```
+
+```c++
+/* Create a egress VLAN Stacking swap entry object:
+ * ------------------------------------------ */
+sai_object_id_t vlan_stacking_oid;
+sai_attribute_t attr;
+vector<sai_attribute_t> vlan_stack_entry_attrs;
+
+attr.id = SAI_VLAN_STACK_ATTR_STAGE;
+attr.value.s32 = SAI_VLAN_STACK_STAGE_EGRESS;
+vlan_stacking_entry_attrs.push(attr);
+
+attr.id = SAI_VLAN_STACK_ATTR_ACTION;
+attr.value.s32 = SAI_VLAN_STACK_ACTION_SWAP
+vlan_stacking_entry_attrs.push(attr);
+
+attr.id = SAI_VLAN_STACK_ATTR_MATCH_TYPE;
+attr.value.s32 = SAI_VLAN_STACK_MATCH_TYPE_OUTER;
+vlan_stacking_entry_attrs.push(attr);
+
+attr.id = SAI_VLAN_STACK_ATTR_ORIGINAL_VLAN_ID;
+attr.value.u16 = 20;
+vlan_stacking_entry_attrs.push(attr);
+
+attr.id = SAI_VLAN_STACK_ATTR_PORT;
+attr.value.oid = 30;
+vlan_stacking_entry_attrs.push(attr);
+
+attr.id = SAI_VLAN_STACK_ATTR_APPLIED_VLAN_ID;
+attr.value.u16 = 10;
+vlan_stacking_entry_attrs.push(attr);
+
+sai_vlan_api->create_vlan_stack(&vlan_stacking_oid, gSwitchId, (uint32_t)vlan_stacking_entry_attrs.size(), vlan_stacking_entry_attrs.data());
+```
+
+# Configuration and management
+
+## CLI/YANG model Enhancements
+
+### CLI
+
+CLI commands for VLAN stacking are not supported yet.
+
+### Yang model
+
+New yang model `sonic-vlan-stacking.yang` is defined to describe VLAN stacking configuration.
+
+```json
+module sonic-vlan-stacking {
+
+    yang-version 1.0;
+
+    namespace "http://github.com/Azure/sonic-vlan-stacking";
+    prefix vlan-stacking;
+
+    import sonic-extension {
+        prefix ext;
+        revision-date 2019-07-01;
+    }
+
+    import sonic-port {
+        prefix port;
+        revision-date 2019-07-01;
+    }
+
+    import sonic-portchannel {
+        prefix lag;
+        revision-date 2019-07-01;
+    }
+
+    import sonic-vlan {
+        prefix vlan;
+        revision-date 2019-07-01;
+    }
+
+    description "VLAN Stacking YANG Module for SONiC OS";
+
+    revision 2021-12-15 {
+        description "First Revision";
+    }
+
+    container sonic-vlan-stacking {
+        container VLAN_STACKING {
+            description "VLAN_STACKING part of config_db.json";
+
+            list VLAN_STACKING_LIST {
+                key "INTERFACE_NAME STAGE VLAN_ID";
+
+                ext:key-regex-configdb-to-yang "^([a-zA-Z0-9_-]+)|([a-zA-Z0-9_-]+)|([0-9]+)$";
+
+                ext:key-regex-yang-to-configdb "<INTERFACE_NAME>|<STAGE>|<VLAN_ID>";
+
+                leaf INTERFACE_NAME {
+                    type union {
+                        type leafref {
+                            path /port:sonic-port/port:PORT/port:PORT_LIST/port:port_name;
+                        }
+                        type leafref {
+                            path /lag:sonic-portchannel/lag:PORTCHANNEL/lag:PORTCHANNEL_LIST/lag:portchannel_name;
+                        }
+                    }
+                }
+
```

Comment:
Please update the field names in lower case for consistency.

Reply:
Sure, next commit will including this.